mico-core/src/main/java/io/github/ust/mico/core/service/GitHubCrawler.java

```diff
@@ -53,48 +57,96 @@ public GitHubCrawler(RestTemplate restTemplate, KubernetesNameNormalizer kuberne
+            ObjectMapper mapper = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
             JsonNode basicInfoJson = mapper.readTree(responseBasicInfo.getBody());
             JsonNode releaseInfoJson = mapper.readTree(responseReleaseInfo.getBody());
 
             String name = basicInfoJson.get("name").textValue();
             String normalizedName = kubernetesNameNormalizer.normalizeName(name);
 
-            return new MicoService()
+            MicoService micoService = new MicoService()
                 .setShortName(normalizedName)
                 .setName(basicInfoJson.get("full_name").textValue())
                 .setVersion(releaseInfoJson.get("tag_name").textValue())
                 .setDescription(basicInfoJson.get("description").textValue())
                 .setServiceCrawlingOrigin(MicoServiceCrawlingOrigin.GITHUB)
                 .setGitCloneUrl(basicInfoJson.get("clone_url").textValue())
                 .setGitReleaseInfoUrl(releaseInfoJson.get("url").textValue());
+            if (!StringUtils.isEmpty(dockerfilePath)) {
+                if (existsFileInGithubRepo(uriBasicInfo, dockerfilePath)) {
+                    micoService.setDockerfilePath(dockerfilePath);
+                } else {
+                    throw new IllegalArgumentException("The dockerfile path must be a valid path relativ the the repository root");
+                }
+            }
+            return micoService;
+        } else {
+            throw new IllegalArgumentException("An error occurred while requesting information about the github repository. Send to requests and got the status codes "
+                + responseBasicInfo.getStatusCode() + " and " + responseReleaseInfo.getStatusCode().is2xxSuccessful());
+        }
 
-        } catch (IOException e) {
-            log.error(e.getStackTrace().toString());
-            log.error("Getting exception '{}'", e.getMessage());
-            throw e;
+    }
+
+    /**
+     * Checks if a file in a github repository exists
+     * @param uriBasicInfo
+     * @param dockerfilePath
+     * @return
+     */
+    private boolean existsFileInGithubRepo(String uriBasicInfo, String dockerfilePath) {
+        ResponseEntity<String> responseDockerFileInfo;
+        UriComponentsBuilder dockerFileUriBuilder = UriComponentsBuilder.fromHttpUrl(uriBasicInfo);
+        UriComponents dockerFileUriComponent = dockerFileUriBuilder.pathSegment(GITHUB_API_CONTENTS).pathSegment(dockerfilePath).build();
+        log.info("Check if the dockerfile exists at {}",dockerFileUriComponent.toString());
+        responseDockerFileInfo = restTemplate.getForEntity(dockerFileUriComponent.toUri(), String.class);
+        if(!responseDockerFileInfo.getStatusCode().is2xxSuccessful()){
+            log.error("Got an error {} while checking if dockerfile exists",responseDockerFileInfo.getStatusCode());
```

Comment:
Is this the normal case if a file does not exist? If yes, then the log statement should be more precisely.